src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,185 @@
+import org.jboss.aerogear.android.store.DataManager;
+import org.jboss.aerogear.android.store.generator.IdGenerator;
+import org.jboss.aerogear.android.store.sql.SQLStore;
+import org.jboss.aerogear.android.store.sql.SQLStoreConfiguration;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import timber.log.Timber;
+
+/**
+ * ModifiedAuthzModule.
+ * <p/>
+ * Performs all related work of generating new key and secret pair, Storing them as account,
+ * Retrieving stored account, Deleting stored account.
+ */
+
+public class SecretStoreAuthzModule implements AuthzModule {
+
+    private Context context;
+    private SQLStore<OAuth2AuthzSession> sessionStore;
+
+    public SecretStoreAuthzModule(Context context) {
+        this.context = context.getApplicationContext();
+        sessionStore = openSessionStore();
+    }
+
+    private SQLStore<OAuth2AuthzSession> openSessionStore() {
+        DataManager.config(AuthEndPoints.STORE, SQLStoreConfiguration.class)
+                .forClass(OAuth2AuthzSession.class)
+                .withContext(context)
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        sessionStore = (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthEndPoints.STORE);
+        sessionStore.openSync();
+        return sessionStore;
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        return storedAccount != null;
+    }
+
+    public void requestAccess(final Activity activity, final Callback<String> callback,
+                              final String username, final String password, final URL url) {
+
+        Exception exception;
+        String json;
+        AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
+            private Exception exception;
+            String json;
+
+            @Override
+            protected HeaderAndBody doInBackground(Void... params) {
+                HeaderAndBody result;
+                try {
+                    HttpProvider provider =
+                            new HttpRestProvider(new URL(url.toString()
+                                    + AuthEndPoints.Endpoints.ACCESS));
+                    provider.setDefaultHeader("Accept", "application/json");
+                    provider.setDefaultHeader("Authorization", "Basic "
+                            + buildLoginData(username, password));
+                    result = provider.post("");
+                    Timber.e("Called with POST to: ", new String(provider.getUrl().toString()));
```

Comment:
`Timber.e` -> sounds like "e" stands for "error", no? Shouldn't it be `Timber.d`?